### ISSUE TITLE
MQTT/XToys integration

### ISSUE THREAD

**Issue by RupertB-ES** [CrashOverride85/zc95#56]
I believe there is a LUA implementation of MQTT -any plans on this or other routes to integrate with XToys?

**Comment by RupertB-ES** [comment]
This is what caught my eye https://github.com/xHasKx/luamqtt

**Comment by CrashOverride85** [comment]
I think the problem's going to be that it depends on luasocket, which depends on a BSD-style sockets API, which LwIP used by the Pico SDK doesn't really provide. Plus I think trying to run Lua scripts of that size/complexity just isn't going to work well (if at all).
I think the way to get mqtt support in Lua would be to add a wrapper around the LwIP MQTT client to expose it in the Lua environment. 

Also I need to have a think about how it would work when running Lua script locally, as at present, the box will only connect to WiFi when in remote access mode.

**Comment by RupertB-ES** [comment]
I guess there are 2 routes - initially could the LUA client just handshake with a python app and let python do the work but in the longer term having the LUA client "phone home" and start the wifi in code would be the better way performance headroom permitting.

**Comment by CrashOverride85** [comment]
Yeah, doing _something_ via the python app might be an option too
